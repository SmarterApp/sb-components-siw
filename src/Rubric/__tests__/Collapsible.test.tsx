import * as React from "react";
import { Collapsible } from "../Collapsible";
import { shallow } from "enzyme";

describe("Collapsible", () => {
  const wrapper = shallow(<Collapsible label="collapsible" />);
  it("expands", () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.findWhere(node => node.type() === "a").simulate("click");
    expect(wrapper).toMatchSnapshot();
  });

  it("collapses", () => {
    wrapper.findWhere(node => node.type() === "a").simulate("click");
    expect(wrapper).toMatchSnapshot();
  });
});
