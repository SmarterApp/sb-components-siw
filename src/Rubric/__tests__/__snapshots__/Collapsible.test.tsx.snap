// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible collapses 1`] = `
<div
  aria-expanded={false}
  role="tab"
>
  <a
    className="collapsible-label link-button"
    href="javascript:"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    ▶ collapsible
  </a>
  <div
    aria-hidden={true}
    className="collapsible-body"
    style={
      Object {
        "display": "none",
      }
    }
  />
</div>
`;

exports[`Collapsible expands 1`] = `
<div
  aria-expanded={false}
  role="tab"
>
  <a
    className="collapsible-label link-button"
    href="javascript:"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    ▶ collapsible
  </a>
  <div
    aria-hidden={true}
    className="collapsible-body"
    style={
      Object {
        "display": "none",
      }
    }
  />
</div>
`;

exports[`Collapsible expands 2`] = `
<div
  aria-expanded={true}
  role="tab"
>
  <a
    className="collapsible-label link-button"
    href="javascript:"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    ▼ collapsible
  </a>
  <div
    aria-hidden={false}
    className="collapsible-body"
    style={Object {}}
  />
</div>
`;
